src/platformdirs/android.py

```diff
@@ -117,23 +117,50 @@ def site_runtime_dir(self) -> str:
 
 
 @lru_cache(maxsize=1)
-def _android_folder() -> str | None:
+def _android_folder() -> str | None:  # noqa: C901, PLR0912
     """:return: base folder for the Android OS or None if it cannot be found"""
-    try:
-        # First try to get a path to android app via pyjnius
-        from jnius import autoclass  # noqa: PLC0415
-
-        context = autoclass("android.content.Context")
-        result: str | None = context.getFilesDir().getParentFile().getAbsolutePath()
-    except Exception:  # noqa: BLE001
-        # if fails find an android folder looking a path on the sys.path
+    result: str | None = None
+    # type checker isn't happy with our "import android", just don't do this when type checking
+    # see https://stackoverflow.com/a/61394121
+    if not TYPE_CHECKING:
+        try:
+            # First try to get a path to android app using python4android (if available)...
+            from android import mActivity  # noqa: PLC0415
+
+            context = cast("android.content.Context", mActivity.getApplicationContext())  # noqa: F821
+            result = context.getFilesDir().getParentFile().getAbsolutePath()
+        except Exception:  # noqa: BLE001
+            result = None
+    if result is None:
+        try:
+            # ...and fall back to using plain pyjnius, if python4android isn't
```

Comment:
Please realign to 120 characters.

Reply:
what do you mean? this is less than 120 chars according to my text editor

Reply:
You should keep adding words to this line until the next word would go over 120.

Reply:
That makes this

```
# ...and fall back to using plain pyjnius, if python4android isn't available or doesn't deliver any useful
# result...
```

And I don't think this is particularly readable.
But if you want this, sure, I'll update the pr.

Reply:
done